packages/utils/src/misc.ts

```diff
@@ -123,6 +128,11 @@ export function formatStorageKeyName(context: string): string {
   return parseContextNames(context).join(COLON);
 }
 
+// -- object ------------------------------------------------ //
+export function isNamespaceEqual(a: SessionTypes.Namespace, b: SessionTypes.Namespace) {
+  return JSON.stringify(a) === JSON.stringify(b);
```

Comment:
This doesn't account for key ordering in the objects, returning false when the namespaces are effectively equal e.g.

```
a = {chains: ["chain"], events: ["event"], methods: ["method"] }
b = {chains: ["chain"],  methods: ["method"], events: ["event"] }

-> returns `false`
```

Reply:
I'd say this is a case for simply adding `lodash.isEqual` to take deep object comparisons off our hands (https://lodash.com/docs/4.17.15#isEqual).

Otherwise it's a matter of manually iterating each key and in turn iterating the string arrays between e.g. `a.chains` and `b.chains` etc